central/apitoken/datastore/datastore_impl.go

```diff
@@ -106,3 +136,34 @@ func (b *datastoreImpl) RevokeToken(ctx context.Context, id string) (bool, error
 	}
 	return true, nil
 }
+
+func (b *datastoreImpl) Search(ctx context.Context, q *v1.Query) ([]search.Result, error) {
+	if err := sac.VerifyAuthzOK(integrationSAC.ReadAllowed(ctx)); err != nil {
+		return nil, err
+	}
+	return b.searcher.Search(ctx, q)
+}
+
+func (b *datastoreImpl) SearchRawTokens(ctx context.Context, q *v1.Query) ([]*storage.TokenMetadata, error) {
+	if err := sac.VerifyAuthzOK(integrationSAC.ReadAllowed(ctx)); err != nil {
+		return nil, err
+	}
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		return b.storage.GetByQuery(ctx, q)
+	}
+	results, err := b.searcher.Search(ctx, q)
```

Comment:
what happens here?  I think `b.searcher` is going to be nil here as it is in Rocks mode and you set the searcher to nil at line 51.

Reply:
It's likely to panic.
Although the chances that this code gets used in rocks/bleve mode are low, I'll make sure an error is returned when not in postgres mode.